### ISSUE TITLE
Segmentation fault. "b-httpworker.0"

### ISSUE THREAD

**Issue by IvRRimum** [bitcoin/bitcoin#24388]
After calling `importaddress` trough HTTP RPC, i get segmentation fault at the end of sync(tracked by progress.

**Expected behavior**
Should complete importaddress

Should complete importaddress

**Actual behavior**
Segmentation fault

**To reproduce**
Just try to call `importaddress` trought wallet RPC in bitcoin testnet.

10/10 times

**System information**
Ubuntu

0.21.1

Digital ocean Ubuntu instance

gdb

Heres log right before crash:
```
2022-02-19T10:11:15Z sending verack (0 bytes) peer=2291
2022-02-19T10:11:15Z receive version message: /Satoshi:0.17.1/: version 70013, blocks=0, us=68.183.75.14:18333, peer=2291
2022-02-19T10:11:15Z socket closed for peer=2290
2022-02-19T10:11:15Z disconnecting peer=2290
2022-02-19T10:11:15Z received: verack (0 bytes) peer=2290
2022-02-19T10:11:15Z sending sendheaders (0 bytes) peer=2290

Thread 11 "b-httpworker.0" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffcffff700 (LWP 1008189)]
```

**Comment by MarcoFalke** [comment]
There is no `loadaddress` RPC. What is `bitcoind -version`?

**Comment by IvRRimum** [comment]
Sorry i meant `importaddress`. I also tested this with the newest version 22

**Comment by IvRRimum** [comment]
> bitcoind -help | head

root@blockchain:~/bitcoin-22.0/bin# ./bitcoind -help | head
Bitcoin Core version v22.0.0
Copyright (C) 2009-2021 The Bitcoin Core developers

Please contribute if you find Bitcoin Core useful. Visit
<https://bitcoincore.org/> for further information about the software.
The source code is available from <https://github.com/bitcoin/bitcoin>.

This is experimental software.
Distributed under the MIT software license, see the accompanying file COPYING
or <https://opensource.org/licenses/MIT>

**Comment by IvRRimum** [comment]
Heres logs on lower logging level:
```
2022-02-19T11:07:15Z UpdateTip: new best=0000000000000224945b2c06ee493b598b7fa6a495f1b0d725a53f0549ce69c3 height=2162250 version=0x20000000 log2_work
=74.650783 tx=62201967 date='2022-02-19T11:06:56Z' progress=1.000000 cache=0.1MiB(489txo)
2022-02-19T11:07:35Z [main] Still rescanning. At block 1280185. Progress=0.277538
2022-02-19T11:08:03Z New outbound peer connected: version: 70016, blocks=2162250, peer=1146 (block-relay-only)
2022-02-19T11:08:35Z [main] Still rescanning. At block 1409789. Progress=0.626206
2022-02-19T11:09:06Z UpdateTip: new best=00000000000009dd2b0f9c0eb915cf1c8c0bb2b9adfdfea9c1bb8e6c38ec470f height=2162251 version=0x20000000 log2_work
=74.650783 tx=62201976 date='2022-02-19T11:08:45Z' progress=1.000000 cache=0.1MiB(502txo)
2022-02-19T11:09:35Z [main] Still rescanning. At block 1663845. Progress=0.870718
2022-02-19T11:09:37Z UpdateTip: new best=0000000000000484ef740ad9cc178c30b3c4e40311f86d461fca3231ce6731b4 height=2162252 version=0x20000000 log2_work
=74.650784 tx=62201977 date='2022-02-19T11:09:06Z' progress=1.000000 cache=0.1MiB(504txo)
Segmentation fault (core dumped)
```

**Comment by IvRRimum** [comment]
Any ideas? @MarcoFalke

**Comment by IvRRimum** [comment]
This is kind of breaking change, any ideas? @MarcoFalke @ldenman @casey @eklitzke

**Comment by IvRRimum** [comment]
In versions V21.1, V22.1 gives me segment.
In version V20.1 It works fine.

**Comment by achow101** [comment]
Probably duplicate of/related to #24335 and #21605.

**Comment by MarcoFalke** [comment]
Closing for now as presumed duplicate of https://github.com/bitcoin/bitcoin/issues/24335 . Please let us know if the issue still persists after https://github.com/bitcoin/bitcoin/issues/24335 is fixed (hopefully in the upcoming 23.0 release).
